src/charm.py

```diff
@@ -99,18 +118,42 @@ def __init__(self, *args):
             self._on_loki_push_api_endpoint_departed,
         )
 
-    def update_alerts_rules(self, _):
+    def _metrics_alerts(self, event):
+        self.update_alerts_rules(
+            event,
+            alerts_func=self._scrape.alerts,
+            reload_func=self._remote_write.reload_alerts,
+            mapping=self.metrics_rules_paths,
+        )
+
+    def _loki_alerts(self, event):
+        self.update_alerts_rules(
+            event,
+            alerts_func=self._loki_provider.alerts,
+            reload_func=self._loki_consumer._reinitialize_alert_rules,
+            mapping=self.loki_rules_paths,
+        )
+
+    def update_alerts_rules(
+        self, _, alerts_func: Any, reload_func: Callable, mapping: RulesMapping
+    ):
         """Copy alert rules from relations and save them to disk."""
-        rules = self._scrape.alerts()
-        shutil.rmtree(self._metrics_rules_dest_path)
-        shutil.copytree(self._metrics_rules_src_path, self._metrics_rules_dest_path)
+        rules = {}
+
+        # MetricsEndpointConsumer.alerts is not @property, but Loki is, so
+        # do the right thing
+        if callable(alerts_func):
```

Comment:
Interesting I did not realize we had this inconsistency between our charms. Not relevant to this PR it leaves me wondering if it worth making `alerts` in Loki and Prometheus consistent, which ever way.